crates/router/src/core/webhooks/webhook_events.rs

```diff
@@ -86,22 +107,36 @@ pub async fn list_initial_delivery_attempts(
 #[instrument(skip(state))]
 pub async fn list_delivery_attempts(
     state: AppState,
-    merchant_id: &str,
-    initial_attempt_id: &str,
+    merchant_id_or_profile_id: String,
```

Comment:
Can't we fetch merchant_account or business_profile during `authenticate_and_fetch` and pass it here?

Reply:
Since we also allow admin API authentication on this endpoint, and admin API authentication is used for endpoints like merchant account create, merchant connector account create, etc. where a merchant account or business profile may not exist beforehand, we cannot always fetch a merchant account or business profile as part of the authentication layer, when using the admin API authentication.